Source/Core/Core/NetPlayServer.cpp

```diff
@@ -1936,6 +1932,10 @@ u16 NetPlayServer::GetPort() const
 void NetPlayServer::SetNetPlayUI(NetPlayUI* dialog)
 {
   m_dialog = dialog;
+
+  // Has to be here otherwise success/failure message won't appear
```

Comment:
Is this the only place where the call to SetupIndex can be placed? It somewhat bothers me that this is done from a function that's called SetNetPlayUI...

Reply:
Yeah I don't like it here either, but I can't think of anywhere else that it can be called at the right time.